pkg/cluster/fetch.go

```diff
@@ -212,6 +253,17 @@ func BuildSpec(ctx context.Context, client Client, cluster *v1alpha1.Cluster) (*
 // BuildSpecFromConfig constructs a cluster.Spec for an eks-a cluster config by retrieving all dependencies objects from the cluster using a kubernetes client.
 func BuildSpecFromConfig(ctx context.Context, client Client, config *Config) (*Spec, error) {
 	bundlesName, bundlesNamespace := bundlesNamespacedKey(config.Cluster)
+	eksaRelease := &v1alpha1release.EKSARelease{}
+	if config.Cluster.Spec.EksaVersion == nil {
+		return nil, fmt.Errorf("Cluster's EksaVersion is nil")
+	}
+	version := string(*config.Cluster.Spec.EksaVersion)
```

Comment:
GenerateEKSAReleaseName takes a string and does string manipulation. I can't have it take v1alpha1.EksaVersion because that would cause a circular dependency.